### PR TITLE
feat(cow-fi): migrate cow-fi to monorepo

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

1. Run `nx g @nx/next:app my-new-app` to generate a new Next app ([ref](https://nx.dev/nx-api/next)) ([commit](https://github.com/cowprotocol/cowswap/commit/9dc1cc14d9652031aee315f0ec259af1a063da18))
2. Just copy and paste files from the old repo ([commit](https://github.com/cowprotocol/cowswap/commit/8be8151c14381659f652ed5099276505524b340e))
3. Fix `<CustomLink>` component ([commit](https://github.com/cowprotocol/cowswap/commit/5aefe18a09908b7ad81dca7d39a4ec657ca0e71d))
4. Fix `<Link>` component usage ([commit](https://github.com/cowprotocol/cowswap/commit/71778973cc86640f77f78e45a3602b3fefcb53cc))
5. Fix compiling errors ([commit](https://github.com/cowprotocol/cowswap/pull/4332/commits/0f43be2ebb972dcbb7d402466357fc9155cc2dd2))

The most of the changes are about the last point:
```
Unhandled Runtime Error
Error: Hydration failed because the initial UI does not match what was rendered on the server.

Warning: Expected server HTML to contain a matching <a> in <a>.

See more info here: https://nextjs.org/docs/messages/react-hydration-error
```

In the cow-fi we used to have code like this:
```
<LinkWithUtm>
  <Button
    target="_blank"
    rel="noopener nofollow"
    ...
  />
</LinkWithUtm>
```

And it is actually `<a><a></a></a>` which is not allowed in the new Next version.
So I changed `<Button>` component and used [`as`](https://styled-components.com/docs/basics) property to make it `span` or `a` depending on the context.

# To Test

image

(the project name will be changed after removing the current one)

1. The whole cow.fi regression testing
2. Pay additional attention on links (internal and external)

### CODE REVIEW

File: apps/cow-fi/COPYRIGHT.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Intellectual Property Notice
+
+Copyright 2021 Gnosis Ltd
+
+Copyrights in this project are retained by contributors. No copyright assignment
+is required to contribute to this project.
+
+Except as otherwise noted (below and/or in individual files), this project is
+licensed under the Apache License, Version 2.0
```
Comment (alfetopito) [inline apps/cow-fi/COPYRIGHT.md:9]: I know, just copied over.
Still, might be good to have legal review this, as we are no longer in 2021 nor Gnosis

File: apps/cow-fi/LICENSE-MIT [hunk 2]
```diff
@@ -0,0 +1,21 @@
+MIT License
+
+Copyright (c) 2021 Gnosis Ltd
```
Comment (alfetopito) [inline apps/cow-fi/LICENSE-MIT:3]: Same thing here, 2021 gnosis

**Comment by elena-zh** [comment]
Hey @shoom3301 , the preview link makes me login into vercel with no success
image

**Review by alfetopito — APPROVED** [review]

**Comment by elena-zh** [comment]
What is the difference between cowswap and swap-dev previews? Btw, cowswap link can't be opened from my side as well (requires login to vercel).
image

**Review by anxolin — APPROVED** [review]
This is amazing! thanks 🙏

**Review by elena-zh — COMMENTED** [review]
Great job!
However, my [question ](https://github.com/cowprotocol/cowswap/pull/4332#issuecomment-2084808494)is still valid

**Comment by alfetopito** [comment]
> What is the difference between cowswap and swap-dev previews? Btw, cowswap link can't be opened from my side as well (requires login to vercel). image

Seems like a bug, maybe fixed already but stuck on the PR description

image

**Comment by shoom3301** [comment]
> > What is the difference between cowswap and swap-dev previews? Btw, cowswap link can't be opened from my side as well (requires login to vercel). image
> 
> Seems like a bug, maybe fixed already but stuck on the PR description
> 
> image

@elena-zh @alfetopito 
Yes, it's an artifact of Github. The project was named `cowswap` initially (because cowfi was already taken and I didn't notice it), then I renamed it to cowfi-new.
So, everything is ok

**Pull request merged**